server/server.go

```
@@ -77,23 +79,28 @@ func (o *OvsdbServer) Serve(protocol string, path string) error {
 	o.ready = true
 	o.readyMutex.Unlock()
 	for {
-		select {
-		case <-o.done:
-			return nil
-		default:
-			conn, err := lis.Accept()
-			if err != nil {
+		conn, err := o.listener.Accept()
+		if err != nil {
```

It seems weird to only check the channel if `Accept` returns an error, and even more weird to ignore the error and return nil

---

Agreed, you get a weird interface and you do weird things around it. Came as a surprise that we need to handle an error when closing the listener gracefully and even more surprise when there is no (that I know of) exported typed error to check for.